rabbit.go

```diff
@@ -475,14 +512,31 @@ func (r *Rabbit) ConsumeOnce(ctx context.Context, runFunc func(msg amqp.Delivery
 			return errors.New("detected nil acknowledger - sent signal to reconnect to RabbitMQ")
 		}
 
-		if err := runFunc(msg); err != nil {
-			return err
+	RETRY:
+		for {
+			if err := runFunc(msg); err != nil {
+				if retry != nil && retry.ShouldRetry() {
+					dur := retry.Duration(retries)
+
+					r.log.Warnf("[Retry %s] error during consume: %s", retry.AttemptCount(), err)
+
+					time.Sleep(dur)
+					retries++
+					continue RETRY
+				}
+
+				r.log.Debug("ConsumeOnce finished - exiting")
+				return err
+			}
+
+			break
 		}
 	case <-ctx.Done():
-		r.log.Warn("stopped via context")
+		r.log.Warn("ConsumeOnce stopped via context")
```

Comment:
`ConsumeOnce stopped via local context`

and 

`ConsumeOnce stopped via global context`

Reply:
Fixed!